napi-inl.h

```diff
@@ -155,26 +155,21 @@ struct AccessorCallbackData {
 ////////////////////////////////////////////////////////////////////////////////
 
 #define NODE_API_MODULE(modname, regfunc)                 \
-  void __napi_ ## regfunc(napi_env env,                   \
-                          napi_value exports,             \
-                          napi_value module,              \
-                          void* priv) {                   \
-    Napi::RegisterModule(env, exports, module, regfunc);  \
+  napi_value __napi_ ## regfunc(napi_env env,             \
+                                napi_value exports) {     \
+    return Napi::RegisterModule(env, exports, regfunc);   \
   }                                                       \
   NAPI_MODULE(modname, __napi_ ## regfunc);
 
 // Adapt the NAPI_MODULE registration function:
 //  - Wrap the arguments in NAPI wrappers.
 //  - Catch any NAPI errors and rethrow as JS exceptions.
-inline void RegisterModule(napi_env env,
-                           napi_value exports,
-                           napi_value module,
-                           ModuleRegisterCallback registerCallback) {
-  details::WrapCallback([&] {
-    registerCallback(Napi::Env(env),
-                     Napi::Object(env, exports),
-                     Napi::Object(env, module));
-    return nullptr;
+inline napi_value RegisterModule(napi_env env,
+                                 napi_value exports,
+                                 ModuleRegisterCallback registerCallback) {
+  return details::WrapCallback([&] {
+    return napi_value(registerCallback(Napi::Env(env),
+                      Napi::Object(env, exports)));
```

Comment:
nit: The last line here should be indented further to align with the previous argument.

Or, the explicit `napi_value()` constructor can be omitted because that conversion can be implicit.